### PR TITLE
remove mutex during numa-cnt calculation

### PR DESCRIPTION
**Pull request by justadogistaken**
A slight optimization.
Removed the mutex while calculating numa count.
Signed-off-by: david.bao <baojn1998@163.com>

### CODE REVIEW

File: pkg/scheduler/plugins/util/util.go [hunk 1]
```diff
@@ -272,29 +274,32 @@ func (nl *NodeLister) List() ([]*v1.Node, error) {
...
-		if reverse {
-			for key := range scores {
-				scores[key] = maxPriority
-			}
+	maxMinDiff := maxCount - minCount
+	for idx := range scores {
+		tmpScore := int64(0)
+
+		if maxMinDiff > 0 {
+			tmpScore = maxPriority * (scores[idx].Score - minCount) / maxMinDiff
```
Comment (shinytang6) [inline pkg/scheduler/plugins/util/util.go:297]: Can you clarify why we need to change the calculation way here?

File: pkg/scheduler/plugins/numaaware/numaaware.go [hunk 2]
```diff
@@ -225,21 +226,21 @@ func filterNodeByPolicy(task *api.TaskInfo, node *api.NodeInfo, nodeResSets map[
...
-	var mx sync.RWMutex
+func getNodeNumaNumForTask(nodeInfo []*api.NodeInfo, resAssignMap map[string]api.ResNumaSets) []api.ScoredNode {
+	nodeNumaCnts := make([]api.ScoredNode, len(nodeInfo))
 	workqueue.ParallelizeUntil(context.TODO(), 16, len(nodeInfo), func(index int) {
 		node := nodeInfo[index]
 		assignCpus := resAssignMap[node.Name][string(v1.ResourceCPU)]
-		mx.Lock()
-		defer mx.Unlock()
-		nodeNumaNumMap[node.Name] = int64(getNumaNodeCntForcpuID(assignCpus, node.NumaSchedulerInfo.CPUDetail))
+		nodeNumaCnts[index] = api.ScoredNode{
```
Comment (shinytang6) [inline pkg/scheduler/plugins/numaaware/numaaware.go:234]: l am ok with the changes here

File: pkg/scheduler/plugins/util/util.go (hunk 1 above)
Comment (shinytang6) [reply pkg/scheduler/plugins/util/util.go:297]: l'm ok with both, it seems that both of the calculation methods are used in k8s )

Comment (justadogistaken) [reply pkg/scheduler/plugins/util/util.go:297]: Yes, So it's better to keep the current normalization.

**Review by shinytang6 — COMMENTED** [review]
Thx, lgtm overall )

File: pkg/scheduler/api/numa_info.go [hunk 3]
```diff
@@ -253,3 +255,9 @@ func (resSets ResNumaSets) Clone() ResNumaSets {
 
 	return newSets
 }
+
+// ScoredNode is the wrapper for node during Scoring.
+type ScoredNode struct {
+	NodeName string
+	Score    int64
```
Comment (huone1) [inline pkg/scheduler/api/numa_info.go:262]: what is the intention of adding this structure?

Comment (justadogistaken) [reply pkg/scheduler/api/numa_info.go:262]: Describe a node and its score. For constructing slice of scoredNode instead of map[string]score

Comment (Thor-wl) [reply pkg/scheduler/api/numa_info.go:262]: I think it is more readable comparing with map.

**Pull request merged**